File: pippip-native-ios/API/Logout.m
//
//  Logout.m
//  pippip-native-ios
//
//  Created by Steve Brenneis on 12/10/17.
//  Copyright © 2017 seComm. All rights reserved.
//

#import "Logout.h"
#import "pippip_native_ios-Swift.h"

@interface Logout ()
{
    SessionState *sessionState;
}

@end

@implementation Logout

- (instancetype)init {
    self = [super init];
    
    sessionState = [[SessionState alloc] init];
    return self;
}

- (NSDictionary*)restPacket {
    
    NSMutableDictionary *packet = [[NSMutableDictionary alloc] init];
    [packet setObject:[NSString stringWithFormat:@"%d", sessionState.sessionId]
               forKey:@"sessionId"];
    [packet setObject:[NSString stringWithFormat:@"%lld", sessionState.authToken]
               forKey:@"authToken"];

    return packet;

}

- (double)restTimeout {
    return 10.0;
}

- (NSString*)restURL {
    return @"https://pippip.secomm.cc/authenticator/logout";
}

@end

File: pippip-native-ios/API/Logout.swift
//
//  Logout.swift
//  pippip-native-ios
//
//  Created by Steve Brenneis on 5/15/18.
//  Copyright © 2018 seComm. All rights reserved.
//

import UIKit

class Logout: NSObject, APIRequestProtocol {

    var path: String  {
        if AccountManager.production() {
            return "/authenticator/logout"
        }
        else {
            return "/logout"
        }
    }
    var timeout: Double = 10.0
    var sessionId: Int32?
    var authToken: Int64?

    var sessionState = SessionState()

    override init() {
        sessionId = sessionState.sessionId
        authToken = sessionState.authToken
        super.init()
    }

    required init?(map: Map) {
        
    }
    
    func mapping(map: Map) {

        sessionId <- map["sessionId"]
        authToken <- map["authToken"]

    }

}
